### PR TITLE
Updating HLT DQM and validation for Hgg paths

### PR DESCRIPTION
**Pull request by mplaner**
This PR adds the offline DQM for HLT as well as the HLT validation for diphoton triggers.

### CODE REVIEW

**Comment by dmitrijus** [comment]
This needs greenlight from @mtosi :)

**Comment by mtosi** [comment]
I think it needs the new updates by bhawna
in https://github.com/cms-sw/cmssw/pull/18971

**Comment by mplaner** [comment]
@mtosi 
Yeah, I would like to wait until that PR is approved and then merge myself to make sure everything still works correctly for us.

I can do this now, if we're confident that there won't be any further changes to #18971.

**Comment by mtosi** [comment]
I think bhawna PR is almost done (just waiting for the integration test)

**Comment by mplaner** [comment]
I updated and tested after merging PR #18971 into my local area.  I've added the additional plots for subleading photon for doublePhoton triggers.

**Comment by mtosi** [comment]
@dmitrijus can we start the integration test, please ?

**Comment by fwyzard** [comment]
please test

**Comment by mtosi** [comment]
what about 
HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v*
?

File: HLTriggerOffline/Higgs/python/hltHiggsValidator_cfi.py [hunk 1]
```diff
@@ -195,20 +195,13 @@
...
-        minCandidates = cms.uint32(2), 
-        ),
-    # seperate directory because it needs a different relval    
-    HggControlPaths = cms.PSet( 
-        hltPathsToCheck = cms.vstring(
-            "HLT_Diphoton30_18_R9Id_OR_IsoCaloId_AND_HE_R9Id_DoublePixelSeedMatch_Mass70_v",
-        ),
+            "HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass90_v",
+            "HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass95_v",
+            "HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v7",
```
Comment (mtosi) [inline HLTriggerOffline/Higgs/python/hltHiggsValidator_cfi.py]: are you sure you want to specify v7 ?!?!?!??!?!

**Comment by mplaner** [comment]
Hi Mia,
You were right on both fronts.  I had originally left off the prescaled low-mass paths as I didn't want to clutter the DQM and we don't get much useful information from the DQM for them.  But we want them in the menu, so I've updated with DQM plots.  And I fixed the typo so we won't be restricted to "v7" for the low mass paths in the validation plots.

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by dmitrijus** [comment]
please test

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by perrotta** [comment]
-1
please rebase (taking in mind that now the master is 93X)

**Comment by kpedro88** [comment]
-1
never merge, only rebase
http://cms-sw.github.io/tutorial-resolve-conflicts.html

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by mplaner** [comment]
Apologies for that error, I had it fixed in my local area, but somehow missed it for the rebase and push.  I've now diffed everything and this commit exactly matches what I've tested.

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by mtosi** [comment]
can you please prepare the back-port for 92x ?

**Comment by mplaner** [comment]
Hi @mtosi,
For the backport, do I submit a separate PR (or are there additional instructions somewhere)?

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
@mplaner for backporting there is a tutorial at http://cms-sw.github.io/tutorial-resolve-conflicts.html#backporting-a-pr 

Basically, you create a new branch based on CMSSW 9.2.4, cherry-pick the commits from this branch into it, test, and make a new PR.

**Comment by fwyzard** [comment]
@mplaner, can you fix the errors, test locally, and submit an update ?

**Comment by mplaner** [comment]
@fwyzard I'll do it, but I'm confused why there were conflicts for this merge.  Locally, everything is up to date and runs without issue (for the nightly build from July 2nd).  I'm traveling today, but should manage to resubmit the PR shortly.

**Comment by mtosi** [comment]
@mplaner, please, rebase

**Comment by mplaner** [comment]
Okay, rebased to current master.  Things should run smoothly assuming there aren't other PRs about to get approved which conflict.

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by dmitrijus** [comment]
The description says "This PR supersedes #18971", but that PR is already merged. So, this PR one cannot supersede that one.
Please update the description.

**Comment by dmitrijus** [comment]
+1

**Comment by mtosi** [comment]
can you point me to the back-port in 92x, please ?
thanks !

File: DQMOffline/Trigger/plugins/PhotonMonitor.cc [hunk 2]
```diff
@@ -167,6 +179,36 @@ void PhotonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
   bookME(ibooker,photonEtaPhiME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
   setTitle(photonEtaPhiME_,"#eta","#phi"); 
 
+  //for diphotons
```
Comment (davidlange6) [inline DQMOffline/Trigger/plugins/PhotonMonitor.cc]: hi @mplaner - if nphotons=1 these don't get used. Please don't fill them in that case.

Comment (davidlange6) [inline DQMOffline/Trigger/plugins/PhotonMonitor.cc]: fill -> book

Comment (mplaner) [inline DQMOffline/Trigger/plugins/PhotonMonitor.cc]: Thanks, that's a good point and has been updated.

**Comment by mplaner** [comment]
@mtosi The 92X version is here: #19612

**Comment by dmitrijus** [comment]
+1

**Comment by fwyzard** [comment]
The failing tests are unrelated to this PR

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
Please merge #19684 which includes this and fixes conflicts.

**Comment by davidlange6** [comment]
merge

**Comment by fwyzard** [comment]
David,
may I very politely ask "what the fuck ?" ?

I spent some time to go over all the HLT DQM PRs that were OK from the implementation point of view, merge all of them taking care of the conflicts, and write an explicit message on GitHub in each of them.

And the first thing you do is merge one of them at random messing up the confluct resolution I have already done.

.Andrea

**Comment by davidlange6** [comment]
think of it as me being totally lost - I guess you propose to close all of the prs, make me redo the review on something too big to review? To be honest, the opposite approach of small one-feature-one-PR pull requests would be better (but perhaps impossible for reasons I've not understood yet)

**Comment by fwyzard** [comment]
No, I propose something much simpler:
  - we leave the individual PRs open for the moment
  - I merge them one at a time in a single PR, leaving the individual commits untouched, adding only the resolution of the merge conflicts and minor fixes
  - we do not re-review the "merged PR", since all the individual changes were already reviewd
  - once the tests pass, you merge the "merged PR"
  - this should automatically set the original PRs as "merged" as well

If you agree with the plan, I will re-do the "merged PR" based on the current 9.3.X.

**Comment by fwyzard** [comment]
The opposite approach (one PR per feature) is what we have been doing so far.
It works well for reviewing the changes, but is a mess for merging PRs that need to modify the same files, as you have seen.

**Comment by davidlange6** [comment]
> On Jul 11, 2017, at 11:58 AM, Andrea Bocci <notifications@github.com> wrote:
> 
> No, I propose something much simpler:
> 
> 	• we leave the individual PRs open for the moment
> 	• I merge them one at a time in a single PR, leaving the individual commits untouched, adding only the resolution of the merge conflicts and minor fixes
> 	• we do not re-review the "merged PR", since all the individual changes were already reviewd
> 	• once the tests pass, you merge the "merged PR"
> 	• this should automatically set the original PRs as "merged" as well
> If you agree with the plan, I will re-do the "merged PR" based on the current 9.3.X.

ok - fine, but then don't merge small PRs into the merged PR until they are reviewed.

**Pull request merged**